aten/src/ATen/native/Blas.cpp

```diff
@@ -82,23 +90,17 @@ TORCH_IMPL_FUNC(addmv_out_cpu)(const Tensor &self, const Tensor &mat, const Tens
   }
 }
 
-Tensor &mv_out(const Tensor &self, const Tensor &vec, Tensor& result) {
+TORCH_IMPL_FUNC(mv_out)(const Tensor &self, const Tensor &vec,  const Tensor& result) {
   //self arg sent to addmv_out cannot be resized
   //here we use result as self argument for addmv, and result is user supplied and can be wrong size
   //it's not a hard error, because we allow resizing result, but it becomes a hard error
   //in addmv, because addmv expects self to satisfy proper conditions
   //to avoid this, supply correctly sized self, its contents doesn't matter because beta is 0
   if (result.dim() > 1 || (result.numel() != self.size(0) || result.numel() !=1)) {
     Tensor self_addmv = at::empty({self.size(0)}, self.options());
-    return at::addmv_out(result, self_addmv, self, vec, 0, 1);
+    at::addmv_out(const_cast<Tensor&>(result), self_addmv, self, vec, 0, 1);
```

Comment:
@ezyang can this be `at::cpu::addmv_out`? `at:cpu::mul_out` amazingly worked with cuda inputs too, so maybe this can work to avoid dispatch?

Reply:
Hm, so at::addmv_out would call addmv's meta and check sizes again? That seems not great.

Reply:
hmm. good point. I guess originally `mv` was calling `addmv_` or `addmv_out` so it doesn't need to differentiate between cpu and GPU impl. 

How about adding `mv_out_cpu` and `mv_out_cuda` and they call `addmv_out_cpu` and `addmv_out_cuda` respectively?

Reply:
at::cpu:addmv_out might just work (for cuda inputs also), otherwise yeah, probably having separate impls dispatched to cuda and cpu might work.

Reply:
> at:cpu::mul_out amazingly worked with cuda inputs too, so maybe this can work to avoid dispatch?

That's just a hack, and if we ever eliminate DispatchStub this would stop working.

If there's a dispatch stub for addmv, calling it directly would be reasonable-ish. If you need a non-size-checked version of functions we can also consider adding another API for that, wouldn't be hard, just need to design it.

Reply:
Nope, there's no dispatch stub for addmv, its dispatch is specified in native_functions.yaml. In this case yes, non-size-checked version would be useful (mv has already run its meta, it already know that everything is fine, what it needs is to directly call addmv_out impl).